### ISSUE TITLE
Activate compiler warnings in tests

### ISSUE THREAD

**Issue by jputcu** [ETLCPP/etl#571]
Enabling warnings on the unit tests would catch a number of issues before releasing, for example #563.

If it is OK, I would like to help introducing some stricter compiler checks and fixing the warnings that are generated?
I will do this in my own branch and afterwards creating a pull request.

**Comment by jwellbelove** [comment]
I made a bit of a start on `feature/enable-gcc-Wall-and-Wextra-compiler-flags`.
Most of the warning in the unit tests seem to be with the CHECK_EQUAL macros with 'comparing signed and unsigned' warnings.

**Comment by jputcu** [comment]
I see they are more related to `unittest++`, perhaps we can include the headers via CMake `target_include_directories` with `SYSTEM`, this hides compiler warnings for that directory.

**Comment by jwellbelove** [comment]
I've tried to add `target_include_directories(etl_tests SYSTEM PRIVATE ${UNITTEST_DIR})` (and various combinations thereof) but it has no effect on the warnings in the unit test header `Checks.h`.

```
/mnt/d/Users/John/Programming/GitHub/etl/test/test_bit.cpp:204:7:   required from here
/mnt/d/Users/John/Programming/GitHub/etl/test/UnitTest++/Checks.h:21:22: warning: comparison of integer expressions of different signedness: ‘const int’ and ‘const unsigned int’ [-Wsign-compare]
   21 |       if (!(expected == actual))
```

**Comment by jwellbelove** [comment]
I've currently got rid of the `Check.h` warnings by adding #pragma diagnostic push/pop into the header.

**Comment by jwellbelove** [comment]
Fixed 20.32.0
